### ISSUE TITLE
GObject warning when upscaling a WebP image

### ISSUE THREAD

**Issue by kleisauke** [jcupitt/libvips#710]
When upscaling a WebP image:
```php
<?php

use Jcupitt\Vips\Image;
use Jcupitt\Vips\Size;

require __DIR__ . '/vendor/autoload.php';

// http://www.gstatic.com/webp/gallery3/5_webp_a.webp
$image = Image::thumbnail(__DIR__ . '/5_webp_a.webp', 320, [
    'auto_rotate' => true,
    'height' => 320,
    'size' => Size::UP
]);

$image->writeToFile(__DIR__ . '/upscale.webp');
```

It will give this GObject warning:
```bash
 (banana:26328): GLib-GObject-WARNING **: value "0" of type 'gint' is invalid or out of range for property 'shrink' of type 'gint'
```

Looks like the `hshrink` and `vshrink` [here](https://github.com/jcupitt/libvips/blob/aeff03b951d1a7a9aa900869675120112cd66644/libvips/resample/thumbnail.c#L193-L196) can be less than 1 (`300 / 320 = 0.9375`), [webpload](https://github.com/jcupitt/libvips/blob/3e808288055491208bd1138d21e2c66145aa5ad3/libvips/foreign/webpload.c#L104) expect that it's more or equal to 1.

**Comment by jcupitt** [comment]
Hello, reproducible with:

```
$ vips crop quagga.webp x.webp 0 0 300 300
$ vipsthumbnail x.webp -s 320 -o y.webp 

(vipsthumbnail:28047): GLib-GObject-WARNING **: value "0" of type 'gint' is invalid or out of range for property 'shrink' of type 'gint'
```

It works fine with jpg, so I suppose it's something to do with webp shrink-on-load being used to upscale. I'll have a look.

Thanks!

**Comment by jcupitt** [comment]
Fixed in master and 8.5. Phew!
